lib/pka.c

```diff
@@ -2168,3 +2168,41 @@ int pka_dsa_signature_verify(pka_handle_t     handle,
         return pka_submit_cmd(handle, user_data, CC_DSA_VERIFY, &operands);
 }
 
+int pka_get_rand_bytes(pka_handle_t  handle,
+                       uint8_t      *buf,
+                       int           buf_len)
+{
+    pka_global_info_t   *gbl_info;
+    pka_local_info_t    *local_info;
+    pka_ring_info_t     *ring_info;
+    pka_dev_trng_info_t  trng_info;
+    uint32_t             avail_descs_num;
+    int                  ret;
+
+    if (handle == NULL || buf == NULL || buf_len <= 0)
+	return 0;
+
+    local_info = (pka_local_info_t *)handle;
+    gbl_info   = local_info->gbl_info;
+
+    ring_info = pka_has_available_ring(gbl_info->rings, gbl_info->rings_cnt, 0,

```

Comment:
I don't think this is correct. Every instance owns a set of rings, we don't have to verify whether it has available rings, because the request would be processed by the shim, not through rings, am I correct? a pointer to a ring is sufficent no need to call "pka_has_available_ring()"